ui/src/shared/components/Dygraph.js

```diff
@@ -10,6 +10,12 @@ import getRange from 'shared/parsing/getRangeForDygraph'
 import {LINE_COLORS, multiColumnBarPlotter} from 'src/shared/graphs/helpers'
 import DygraphLegend from 'src/shared/components/DygraphLegend'
 
+const hasherino = (str, len) =>
```

Comment:
Yeah, that makes sense because you're just providing it with numbers, and it's meant to work over strings, not just numbers. A number's string char code is going to provide for a very limited range of inputs that this hash function is not designed to provide a great deal of entropy for. However, the expected inputs of longer strings of typical series names, should be fine for what we're doing.

I did make one change, and that was to the fact that I'm modulo-ing by the length in the reducer, and I don't really need to do that. A string would literally have to have a length of `137438953471` to overflow this hashing function, so I'm not going to optimize for that scenario. That is calculated by the number of UTF-16 char codes provided by String.prototype.charCodeAt. I got this number from `Number.MAX_SAFE_INTEGER / 2 ** 16`